pkg/ingester/ingester.go

```diff
@@ -1677,6 +1681,13 @@ func (i *Ingester) QueryStream(req *client.QueryRequest, stream client.Ingester_
 			level.Debug(spanlog).Log("msg", "using executeChunksQuery")
 			numSeries, numSamples, err = i.executeChunksQuery(ctx, db, int64(from), int64(through), matchers, shard, stream)
```

Comment:
(this is probably just me not knowing too much about context cancelations) Do we need to pass in a new context here (as in with `context.Background()`)? I'm unclear about how that cancel signal gets propagated; will things that check `ctx.Err()` and `ctx.Done()` find that the context is canceled even if we set `err = nil` below?

Reply:
Never mind, I think I get it now - disregard!